aggregator_api/src/lib.rs

```diff
@@ -0,0 +1,897 @@
+            HpkeKemId::X25519HkdfSha256,
+            HpkeKdfId::HkdfSha256,
+            HpkeAeadId::Aes128Gcm,
+        )]);
+
+        let task = Task::new(
+            /* task_id */ random(),
+            /* aggregator_endpoints */ req.aggregator_endpoints,
+            /* query_type */ req.query_type,
+            /* vdaf */ req.vdaf,
+            /* role */ req.role,
+            /* vdaf_verify_keys */ vdaf_verify_keys,
+            /* max_batch_query_count */ req.max_batch_query_count,
+            /* task_expiration */ task_expiration,
+            /* report_expiry_age */
+            Some(Duration::from_seconds(3600 * 24 * 7 * 2)), // 2 weeks
+            /* min_batch_size */ req.min_batch_size,
+            /* time_precision */ time_precision,
+            /* tolerable_clock_skew */
+            Duration::from_seconds(60), // 1 minute,
+            /* collector_hpke_config */ req.collector_hpke_config,
+            /* aggregator_auth_tokens */ Vec::from([random()]),
+            /* collector_auth_tokens */ Vec::from([random()]),
+            /* hpke_keys */ hpke_keys,
+        );
+        let task = match task {
+            Ok(task) => Arc::new(task),
+            Err(_) => return conn.with_status(Status::BadRequest).halt(),
+        };
+
+        conn_try!(
+            self.ds
+                .run_tx_with_name("post_task", |tx| {
+                    let task = Arc::clone(&task);
+                    Box::pin(async move { tx.put_task(&task).await })
+                })
+                .await,
+            conn
+        );
+
+        conn.with_json(&TaskResp::from(task.as_ref()))
+    }
+
+    async fn handle_get_task(&self, conn: Conn) -> Conn {
+        let task_id = TaskId::from_str(conn_unwrap!(conn.param("task_id"), conn));
+        let task_id = match task_id {
+            Ok(task_id) => task_id,
+            Err(_) => return conn.with_status(Status::BadRequest).halt(),
+        };
+
+        let task = conn_try!(
+            self.ds
+                .run_tx_with_name("get_task", |tx| {
+                    Box::pin(async move { tx.get_task(&task_id).await })
+                })
+                .await,
+            conn
+        );
+        let task = match task {
+            Some(task) => task,
+            None => return conn.with_status(Status::NotFound).halt(),
+        };
+
+        conn.with_json(&TaskResp::from(&task))
+    }
+
+    async fn handle_delete_task(&self, conn: Conn) -> Conn {
+        let task_id = TaskId::from_str(conn_unwrap!(conn.param("task_id"), conn));
+        let task_id = match task_id {
+            Ok(task_id) => task_id,
+            Err(_) => return conn.with_status(Status::BadRequest).halt(),
+        };
+
+        let rslt = self
+            .ds
+            .run_tx_with_name("delete_task", |tx| {
+                Box::pin(async move { tx.delete_task(&task_id).await })
+            })
+            .await;
+        match rslt {
+            Ok(()) => conn.ok(Body::default()),
```

Comment:
I think HTTP 204 No Content is more conventional, especially if we don't want to provide a response body: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE